tiledb/sm/query/readers/reader_base.cc

```diff
@@ -277,8 +331,8 @@ bool ReaderBase::include_timestamps(const unsigned f) const {
       array_->timestamp_start(), array_->timestamp_end_opened_at());
   auto dups = array_schema_.allows_dups();
 
-  return frag_has_ts &&
-         (user_requested_timestamps_ || partial_overlap || !dups);
+  return frag_has_ts && (user_requested_timestamps_ || partial_overlap ||
+                         !dups || timestamps_needed_for_deletes_[f]);
```

Comment:
Minor style suggestion: could use `timestamps_needed = timestamps_needed_for_deletes[f]` on its own line and use `|| timestamps_needed` here, so that the array lookup is clearly delineated (rather than at the end of a compound boolean statement).